setup

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# -e (-o errexit): exit immediately on error
+# -u (-o nounset): treat unset variables as erros
+# -o pipefail: propagate intermediate errors in pipelines
+set -euo pipefail
+
+main() {
+  print_step "Verifying that pip is available"
+  pip=$(command -v pip3 || command -v pip || echo -n "")
+  if [ -z "$pip" ]; then
+    print_error "Cannot find pip, aborting."
+    exit 1
+  fi
+
+  print_step "Installing and initializing pipenv"
+  $pip install pipenv
+
+  print_step "Installing Python dependencies"
+  pipenv install
+
+  print_step "Generating Ansible Vault secret"
+  pwgen -1 24|tee ~/.forem_selfhost_ansible_vault_password
+
+  print_step "Copying example inventory"
+  cp -v inventory/example/setup.yml inventory/forem/setup.yml
```

Comment:
Another spot where we need to be idempotent. We don't want to clobber their old inventory if the script is run again.

Reply:
[GNU `cp`](https://www.gnu.org/software/coreutils/manual/html_node/cp-invocation.html) has a `--no-clobber` that does this.

It looks like bsd and linux both accept `-n` for this meaning case. I have to assume the cp in mac os behaves like [freebsd](https://www.unix.com/man-page/FreeBSD/1/cp/)'s, but [posix](https://www.man7.org/linux/man-pages/man1/cp.1p.html) only requires cp to provide the `-i` interactive option to prompt if the file is observed (I doubt whether we want to make that interactive here).

Reply:
I prob would just wrap it in an if statement that checks if the file exists and if so it does a noop and prints a warning.

Reply:
I didn't see this discussion before I made the changes but FWIW I pretty much did what @jdoss suggested mostly because `cp -n` would just do nothing if the file already exists and I'd rather output an info statement.